File: Recyclerview/app/src/main/java/com/example/recyclerview/MainActivity.java
package com.example.recyclerview;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.recyclerview.databinding.ActivityMainBinding;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private ArrayList<MainBata> arrayList;
    private MainAdapter mainAdapter;
    private RecyclerView recyclerView;
    private EditText editText;
    private LinearLayoutManager linearLayoutManager;
    private ActivityMainBinding binding;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        ActionBar actionBar = getSupportActionBar();
        actionBar.hide();

        recyclerView = (RecyclerView)findViewById(R.id.rv);
        linearLayoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(linearLayoutManager);

        arrayList = new ArrayList<>();
        mainAdapter = new MainAdapter(arrayList);
        recyclerView.setAdapter(mainAdapter);

        Button add = (Button)findViewById(R.id.add);
        EditText editText = (EditText)findViewById(R.id.edittext);

        add.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                MainBata mainBata = new MainBata(R.drawable.ic_launcher_foreground,editText.getText(),"안녕");
                arrayList.add(mainBata);
                mainAdapter.notifyDataSetChanged();
                editText.setText("");
            }
        });
    }
}

File: RecyclerView_kt/app/src/main/java/com/example/recyclerview_kt/MainActivity.kt
package com.example.recyclerview_kt

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import androidx.constraintlayout.helper.widget.Carousel
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val rv = findViewById<RecyclerView>(R.id.rv)

        var itemlist = arrayListOf<Data>(
            Data("rladmsdh", "student", 12),
            Data("rladmsdh", "student", 13),
            Data("rladmsdh", "student", 14),
            Data("rladmsdh", "student", 15)
        )

        rv.layoutManager = LinearLayoutManager(this,LinearLayoutManager.VERTICAL,false)
        rv.setHasFixedSize(true)
        rv.adapter = Adapter(itemlist)
    }
}